src/leader.py

```diff
@@ -142,14 +143,23 @@ def lead(self):
         Moves the leader vehicle based on controller input.
         """
         code, position = self._controller.get_input()
-        if code == 'right_trigger':
-            self.set_speed(position)
-            self.drive()
-        if code == 'left_trigger':
-            self.set_speed(position)
-            self.reverse()
-        if code == 'dpad_left_right' or code == 'left_stick_x':
-            self.turn(code, position)
+        if code is not None:
+            self._mode = code
```

Comment:
style: put a blank line after this. I thought it was a part of the if elif block